src/api/focus.js

```diff
@@ -111,114 +111,112 @@ class Focus {
-        bootloader.vendorId
-      );
-
-      for (const device of deviceList) {
-        const pid = device.deviceDescriptor.idProduct,
-          vid = device.deviceDescriptor.idVendor;
-
-        if (pid == bootloader.productId && vid == bootloader.vendorId) {
-          logger("focus").info("bootloader found", {
-            device: bootloader,
-            function: "waitForDFUBootloader",
-          });
-          return true;
-        }
-      }
+    const deviceList = await ipcRenderer.invoke(
+      "usb.scan-for-devices",
+      bootloader.productId,
+      bootloader.vendorId
+    );
 
-      logger("focus").debug("bootloader not found, waiting 2s", {
-        function: "waitForDFUBootloader",
-      });
-      await delay(2000);
+    for (const device of deviceList) {
+      const pid = device.deviceDescriptor.idProduct,
+        vid = device.deviceDescriptor.idVendor;
+
+      if (pid == bootloader.productId && vid == bootloader.vendorId) {
+        const newPort = Object.assign({}, device);
+        newPort.focusDeviceDescriptor = focusDeviceDescriptor;
+        newPort.focusDeviceDescriptor.bootloader = true;
+
+        logger("focus").info("bootloader found", {
+          device: bootloader,
+          function: "checkNonSerialBootloader",
+        });
+        return newPort;
+      }
     }
 
-    logger("focus").warn("bootloader not found after 10 attempts", {
-      function: "waitForDFUBootloader",
+    logger("focus").debug("bootloader not found", {
+      function: "checkNonSerialBootloader",
+      device: bootloader,
     });
-    return false;
+
+    return null;
   }
 
-  async waitForBootloader(focusDeviceDescriptor) {
+  async checkBootloader(focusDeviceDescriptor) {
     if (!focusDeviceDescriptor.usb.bootloader) {
-      this.logger.warn("No bootloader defined in the device descriptor", {
+      this.logger().warn("No bootloader defined in the device descriptor", {
         descriptor: focusDeviceDescriptor,
       });
-      return null;
+      return false;
     }
-    logger("focus").info("waiting for bootloader", {
+    logger("focus").info("checking bootloader presence", {
       descriptor: focusDeviceDescriptor,
     });
 
-    if (focusDeviceDescriptor.usb.bootloader.type == "dfu") {
-      return await this.waitForDFUBootloader(focusDeviceDescriptor);
+    if (focusDeviceDescriptor.usb.bootloader.type == "non-serial") {
+      return await this.checkNonSerialBootloader(focusDeviceDescriptor);
     } else {
-      return await this.waitForSerialBootloader(focusDeviceDescriptor);
+      return await this.checkSerialBootloader(focusDeviceDescriptor);
     }
   }
 
   async reconnectToKeyboard(focusDeviceDescriptor) {
     logger("focus").info("reconnecting to keyboard", {
       descriptor: focusDeviceDescriptor,
     });
-    const d = await this.waitForSerialDevice(
+    const d = await this.checkSerialDevice(
       focusDeviceDescriptor,
       focusDeviceDescriptor.usb
     );
+    if (!d) return false;
```

Comment:
what is "d" here? is it "device connected"? is it "device available"? can you use a longer name?